### ISSUE TITLE
[Due for payment 2025-02-13] [$250] mWeb - Workspace - Input is not auto-focused and keyboard doesn't open when creating a tag

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53208]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.67-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** htad26+tttyyu@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Login to new dot with an account
2. Create a new workspace
3. Go to more features > Tags > Enable tags
4. Navigate to tags page > Add tag

## Expected Result:
Tag creation page is opened, the name input is automatically focused and keyboard is opened

## Actual Result:
The name input isn't automatically focused and the keyboard is not opened.
Sometimes the focus and the keyboard appear with a delay

## Workaround:
Unknown

## Platforms:

- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
 
https://github.com/user-attachments/assets/e3fab1af-c01b-44f4-9776-13616a26d0a0

  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021863569158386608990</li>
        <li>Upwork Job ID: 1863569158386608990</li>
        <li>Last Price Increase: 2024-12-09</li>
<li>Automatic offers: </li>
<ul>
<li>linhvovan29546 | Contributor | 105879100</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sonialiap</details>

**Comment by Shahidullah-Muffakir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When user opens the Tags creating page, the input is not focused and the keyboard is not opened.
### What is the root cause of that problem?
https://github.com/Expensify/App/blob/c62ab78b34dbc993c241277b4ee525ec6277b471/src/pages/workspace/tags/WorkspaceCreateTagPage.tsx#L92-L100
here we did not pass :
 autoFocus
  shouldDelayFocus

### What changes do you think we should make in order to solve the problem?

pass these   

  autoFocus
   shouldDelayFocu
1. autoFocus: it will focus the input
2. shouldDelayFocus: it will open the keyboard.

```diff
<InputWrapper 
     InputComponent={TextInput} 
+     autoFocus
+    shouldDelayFocus
```

**Comment by Shahidullah-Muffakir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When user opens the Tags creating page, the input is not focused and the keyboard is not opened.
### What is the root cause of that problem?
https://github.com/Expensify/App/blob/c62ab78b34dbc993c241277b4ee525ec6277b471/src/pages/workspace/tags/WorkspaceCreateTagPage.tsx#L92-L100
here we did not pass :
 autoFocus
  shouldDelayFocus

### What changes do you think we should make in order to solve the problem?

pass these   

  autoFocus
   shouldDelayFocu
1. autoFocus: it will focus the input
2. shouldDelayFocus: it will open the keyboard.

```diff
<InputWrapper 
     InputComponent={TextInput} 
+     autoFocus
+    shouldDelayFocus
```

**Comment by CyberAndrii** [comment]
It's the same issue as https://github.com/Expensify/App/issues/53314 and will be fixed by proposals from there. @sonialiap can we put this on hold and retest later?

**Comment by CyberAndrii** [comment]
I don't think your proposals have the correct RCA and solution. We already use `useAutoFocusInput()` [here](https://github.com/Expensify/App/blob/c62ab78b34dbc993c241277b4ee525ec6277b471/src/pages/workspace/tags/WorkspaceCreateTagPage.tsx#L33) which should handle the focus. However, there's a bug that prevents `InteractionManager.runAfterInteractions` callbacks from executing. Also, `useAutoFocusInput()` is preferred over `autoFocus`.

**Comment by CyberAndrii** [comment]
I did some more testing and it appears that specifically in this case `InteractionManager` callbacks are being blocked by a playing lottie animation on the previous page. If I disable it [here](https://github.com/Expensify/App/blob/3cc88f5a83a979c5e588e01065083a3d714e69da/src/components/Lottie/index.tsx#L100-L118) by setting `autoPlay={false}` then autofocus works again.

cc @Themoonalsofall @Shahidullah-Muffakir if you want to investigate this further and update your proposals.

I wasn't able to reproduce this issue on a Pixel 8 Pro emulator so I tried various Browserstack devices and the results are as follows

**Reproducible**
- Galaxy A10
- Redmi Note 12 4G
- Redmi Note 8

**Not reproducible**
- Galaxy S22 Ultra
- Galaxy S24 Ultra
- Pixel 6 Pro
- Pixel 4
- OnePlus 11R

**Comment by mollfpr** [comment]
Not overdue 👀

**Comment by sonialiap** [comment]
I think this one might be like https://github.com/Expensify/App/issues/53346 and waiting on [React Native 0.76 upgrade](https://github.com/Expensify/App/issues/48911)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by linhvovan29546** [comment]
> I think this one might be like #53346 and waiting on [React Native 0.76 upgrade](https://github.com/Expensify/App/issues/48911)

I think this is not correct. [https://github.com/Expensify/App/issues/48911](https://github.com/Expensify/App/issues/48911) only applies to native, while this issue appears on mWeb.

**Comment by linhvovan29546** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
mWeb - Workspace - Input is not auto-focused and keyboard doesn't open when creating a tag
### What is the root cause of that problem?
On the "Tags" page, a Lottie animation is displayed when the tags is empty
https://github.com/Expensify/App/blob/8ca44dac9fbd0db0d8c57694f0a46a003b3245df/src/pages/workspace/tags/WorkspaceTagsPage.tsx#L407-L418
Then, when navigating to the "Add Tag" page, we have a hook is used to auto-focus the input, the focus is triggered inside `InteractionManager.runAfterInteractions`:
https://github.com/Expensify/App/blob/c62ab78b34dbc993c241277b4ee525ec6277b471/src/pages/workspace/tags/WorkspaceCreateTagPage.tsx#L33
However, the Lottie animation continues running in the background, preventing `runAfterInteractions` from executing.
https://github.com/Expensify/App/blob/c62ab78b34dbc993c241277b4ee525ec6277b471/src/hooks/useAutoFocusInput.ts#L28-L34

### What changes do you think we should make in order to solve the problem?
We have logic to pause the animation when the screen blurs, but this does not apply to `isSideModalNavigator`. Because, on desktop or larger screens, a backdrop is present, and the previous screen remains visible.
https://github.com/Expensify/App/blob/c62ab78b34dbc993c241277b4ee525ec6277b471/src/components/Lottie/index.tsx#L76-L77
We should pause the `Lottie animation` on blur for `mWeb`. Because, on mWeb, the previous page is not visible as the sidebar renders in full-page mode
https://github.com/Expensify/App/blob/c62ab78b34dbc993c241277b4ee525ec6277b471/src/components/Lottie/index.tsx#L76-L77
```js
            if (!isSideModalNavigator(targetRouteName) || isMobile()) {
                setHasNavigatedAway(true);
            }
```

<table align="center">
  <tr>
    <td align="center"><strong>Before</strong></td>
    <td align="center"><strong>After</strong></td>
  </tr>
  <tr>
    <td>
      <kbd>
<video src="https://github.com/user-attachments/assets/35e2932d-cafd-478c-8ba5-a5786ed07e4c" width="100%" controls></video>
</kbd>
    </td>
    <td>
      <kbd>
<video src="https://github.com/user-attachments/assets/e214761f-d843-40a8-aa4a-9ae6a132e2f6" width="100%" controls></video>
  </kbd>
    </td>
  </tr>
</table>

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?
N/A

### What alternative solutions did you explore? (Optional)
N/A

**Comment by linhvovan29546** [comment]
> I think this is not correct. [React Native 0.76 upgrade](https://github.com/Expensify/App/issues/48911) only applies to native, while this issue appears on mWeb.

Pinging @sonialiap and @mollfpr, in case you may have missed my comment

**Comment by mollfpr** [comment]
@linhvovan29546 I am unable to reproduce the issue on my Samsung phone or with the simulator. What might be causing this?

https://github.com/user-attachments/assets/4eb1e472-26e9-4c23-85b0-d707ebcea1de

**Comment by linhvovan29546** [comment]
> It will be hard if we can't constantly reproduce the issue on the affected platform and I can't confirm if the solution works.

Honestly, I can still reproduce this issue on staging with my device, but it seems challenging since you are unable to reproduce it on your device.

**Comment by mollfpr** [comment]
@linhvovan29546 How can I emulate the high CPU usage on the Android emulator? I try to create an AVD with CPU core of 1 and RAM 512MB, but can't reproduce the issue.

**Comment by mollfpr** [comment]
@rayane-djouah Yup, please take over the issue 🙏

**Comment by MonilBhavsar** [comment]
Is this on HOLD?

**Comment by MonilBhavsar** [comment]
Okay thanks! Took it off HOLD. I will review this tomorrow

**Comment by MonilBhavsar** [comment]
> However, the Lottie animation continues running in the background, preventing runAfterInteractions from executing.

Curious to know why it only happens on specific devices, and is this the only place where this issue happens, I don't think so. As we use animations at many places in the App

**Comment by linhvovan29546** [comment]
> Curious to know why it only happens on specific devices.

I'm not sure why this happens on specific devices. I think it may be related to the device's hardware as  some of the affected devices can be found here: https://github.com/Expensify/App/issues/53208#issuecomment-2509803139 (Thank you to @CyberAndrii for investigating!)

> and is this the only place where this issue happens.

The previous screen of the create tag page is the [workspace settings tag page](https://github.com/Expensify/App/blob/8ca44dac9fbd0db0d8c57694f0a46a003b3245df/src/pages/workspace/tags/WorkspaceTagsPage.tsx#L407-L418), which includes a Lottie animation. And on [the create tag page,](https://github.com/Expensify/App/blob/c62ab78b34dbc993c241277b4ee525ec6277b471/src/pages/workspace/tags/WorkspaceCreateTagPage.tsx#L33) we use `useAutoFocusInput` to focus the input field, as mentioned in my [proposal](https://github.com/Expensify/App/issues/53208#issuecomment-2573154531).

> As we use animations at many places in the App.

For many places in the app, I’ve searched thoroughly and noticed that they are not the same as this case.

In summary, my proposal covers the missing case in mWeb https://github.com/Expensify/App/pull/48444, where the `LEFT_MODAL_NAVIGATOR` and `RIGHT_MODAL_NAVIGATOR` are displayed as full-screen views. For better performance, we should pause the lottie animation when a full-screen view is displayed

cc @MonilBhavsar

**Comment by MonilBhavsar** [comment]
Thanks for clarifying 👍

**Comment by sonialiap** [comment]
Payment summary:
- @rayane-d $250 [offer sent](https://www.upwork.com/nx/wm/offer/106113859) in upwork
- @linhvovan29546 $250 paid in upwork

**Comment by sonialiap** [comment]
@rayane-d updated payment summary to reflect payment in ND :)

**Comment by garrettmknight** [comment]
$250 approved for @rayane-d
